### PR TITLE
webserver: set content-type explicitly for js and css files

### PR DESCRIPTION
**Pull request by buck54321**
Because Windows can't seem to figure it out?

### CODE REVIEW

**Comment by chappjc** [comment]
Pretty wild that the issue was because the CLASSES_ROOT .js registry key was explicitly set to text/plain.  Definitely something we should be immune to.

File: client/webserver/webserver.go [hunk 1]
```diff
@@ -262,10 +262,10 @@ func New(core clientCore, addr string, logger dex.Logger, reloadHTML bool) (*Web
 	})
 
 	// Files
-	fileServer(mux, "/js", filepath.Join(siteDir, "dist"))
-	fileServer(mux, "/css", filepath.Join(siteDir, "dist"))
-	fileServer(mux, "/img", filepath.Join(siteDir, "src/img"))
-	fileServer(mux, "/font", filepath.Join(siteDir, "src/font"))
+	fileServer(mux, "/js", filepath.Join(siteDir, "dist"), "text/javascript")
```
Comment (chappjc) [inline client/webserver/webserver.go:265]: "application/javascript", right?

File: client/webserver/webserver.go [hunk 2]
```diff
@@ -449,6 +449,15 @@ func fileServer(r chi.Router, pathRoot, fsRoot string) {
 			return
 		}
 
+		// Ran into a Windows quirk where net was setting the content-type
+		// incorrectly based on a bad registry value or something. This should
+		// prevent that. It's most important for javascript files, because we
+		// add a nosniff header and the browser would refust to execute a js
```
Comment (chappjc) [inline client/webserver/webserver.go:455]: refust

File: client/webserver/webserver.go (hunk 1 above)
Comment (buck54321) [reply client/webserver/webserver.go:265]: Apparently not. https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types

**Review by chappjc — APPROVED** [review]

File: client/webserver/webserver.go (hunk 1 above)
Comment (chappjc) [reply client/webserver/webserver.go:265]: Also ref: https://github.com/golang/go/commit/99df76f0869f00ad9ba53e3aecc22d625ae23de1 and https://github.com/golang/go/issues/32351

**Review by chappjc — APPROVED** [review]
Working as expected.  OS misconfiguration no longer breaks browser execution of js.  Tested with a bad windows registry setting for .js content type set to text/plain and it is still correctly served to frontend as text/javascript.

**Pull request merged**